nMessenger/Source/Messenger/Components/NMessenger.swift

```diff
@@ -102,30 +102,30 @@ public class NMessenger: UIView {
     }
     
     /** Creates the messenger tableview and sets defaults*/
-    private func setupView() {
+    fileprivate func setupView() {
         
-        self.messengerNode.view.keyboardDismissMode = .OnDrag
-        self.addSubview(messengerNode.view)
+        self.messengerNode?.view.keyboardDismissMode = .onDrag
+        self.addSubview((messengerNode?.view)!)
         
-        messengerNode.delegate = self
-        messengerNode.dataSource = self
+        messengerNode?.delegate = self
+        messengerNode?.dataSource = self
         
-        messengerNode.view.setTuningParameters(ASRangeTuningParameters(leadingBufferScreenfuls: 2, trailingBufferScreenfuls: 1), forRangeType: .Display)
+        messengerNode?.view.setTuningParameters(ASRangeTuningParameters(leadingBufferScreenfuls: 2, trailingBufferScreenfuls: 1), for: .display)
         
-        messengerNode.view.separatorStyle = UITableViewCellSeparatorStyle.None
-        messengerNode.view.allowsSelection = false
-        messengerNode.view.showsVerticalScrollIndicator = false
-        messengerNode.view.automaticallyAdjustsContentOffset = true
+        messengerNode?.view.separatorStyle = UITableViewCellSeparatorStyle.none
+        messengerNode?.view.allowsSelection = false
+        messengerNode?.view.showsVerticalScrollIndicator = false
+        messengerNode?.view.automaticallyAdjustsContentOffset = true
     }
     
     //MARK: Public functions
     
     /** Override layoutSubviews to update messenger table frame*/
-    override public func layoutSubviews() {
+    override open func layoutSubviews() {
         super.layoutSubviews()
         //update frame
-        messengerNode.frame = self.bounds
-        messengerNode.view.separatorStyle = UITableViewCellSeparatorStyle.None
+        messengerNode?.frame = self.bounds
+        messengerNode?.view.separatorStyle = UITableViewCellSeparatorStyle.none
```

Comment:
Why is `messengerNode` an optional now? I'd prefer to fix this since it ripples down to so many functions.